zkvm/src/signature/multikey.rs

```diff
@@ -0,0 +1,58 @@
+use crate::signature::VerificationKey;
+use crate::transcript::TranscriptProtocol;
+use curve25519_dalek::ristretto::RistrettoPoint;
+use curve25519_dalek::scalar::Scalar;
+use merlin::Transcript;
+
+#[derive(Clone)]
+pub struct Multikey {
+    transcript: Transcript,
+    aggregated_key: VerificationKey,
+}
+
+impl Multikey {
+    pub fn new(pubkeys: Vec<VerificationKey>) -> Option<Self> {
+        // Create transcript for Multikey
+        let mut transcript = Transcript::new(b"ZkVM.aggregated-key");
+        transcript.commit_u64(b"n", pubkeys.len() as u64);
+
+        // Hash in pubkeys
+        // L = H(X_1 || X_2 || ... || X_n)
+        for X_i in &pubkeys {
+            transcript.commit_point(b"P", &X_i.0);
+        }
+
+        let mut multikey = Multikey {
+            transcript,
+            aggregated_key: VerificationKey(RistrettoPoint::default().compress()),
```

Comment:
there's `curve25519_dalek::constants::RISTRETTO_BASEPOINT_COMPRESSED` to avoid compression cost.

Reply:
nice!